src/meili/topk_search.cc

```diff
@@ -3,14 +3,26 @@
 namespace valhalla {
 namespace meili {
 
+float EnlargedEmissionCostModel::operator()(const StateId& stateid) {
+  // If this was a removed state bail
+  const auto& original_stateid = evs_.GetOrigin(stateid);
```

Comment:
what if `original_stateid` is invalid? i.e. stateid is origin.

Reply:
if its invalid `IsRemoved` will return false, we will then search for the unordinal id in the cache, if its not found, the invalid original id gets sent to `calculate_cost` which will see its invalid and not use it

Reply:
It won't return the cost for `stateid` if `stateid` is an origin, will it? if it's an origin, then `original_stateid` will be invalid, and then we get -1 from `calculate_cost`.

Reply:
if `stateid` is an origin, then if its not marked as redundant it will look it up in the cache. if its not in the cache it will call `calculate_cost`, see the origin is invalid and then use the model to get the cost

Reply:
I get it. You just check if the original stateid is removed or not.

Reply:
yep!